libff/algebra/curves/curve_serialization.tcc

```diff
@@ -134,8 +134,18 @@ public:
         mp_limb_t flags;
         field_read_with_flags<encoding_binary, Form>(group_el.X, flags, in_s);
         if (0 == (flags & 0x2)) {
-            group_el.Y = curve_point_y_at_x<GroupT>(group_el.X);
 
+            Fq x = group_el.X;
+            Fq x_squared = x * x;
+            Fq x_cubed = x_squared * x;
+            Fq y_squared = x_cubed + (GroupT::coeff_a * x) + GroupT::coeff_b;
+            if ((y_squared ^ Fq::euler) != Fq::one()) {
```

Comment:
Fair point. Such checks certainly have costs on the efficiency during normal execution. We can also surround them within ifdef DEBUG.

Reply:
> We can also surround them within ifdef DEBUG.

Yes, perhaps that's the best thing for now.

In general, I prefer to avoid `DEBUG` changing the actual code behaviour, and reserve it for sanity checking or maybe dumping out extra information.  Verifying incoming data is something that a given bit of code should either do or not do, depending on the application, but until we add a parameter for element checking maybe using `DEBUG` is the best tradeoff.

Flagging issue #55, which deals with this specifically (would you mind adding a TODO pointing to this in the code?) and #70 which is tangentially related.  These are probably things we should consider doing soon.  #70 will also help with debugging going forward.

Reply:
All done.